browser/extensions/api/brave_rewards_api.cc

```diff
@@ -75,6 +75,166 @@ BraveRewardsOpenBrowserActionUIFunction::Run() {
+
+  if (!info) {
+    Respond(
+        TwoArguments(std::make_unique<base::Value>(result), std::move(dict)));
+    return;
+  }
+
+  dict->SetStringKey("publisher_key", info->id);
+  dict->SetStringKey("name", info->name);
+  dict->SetIntKey("percentage", info->percent);
+  dict->SetIntKey("status", info->status);
+  dict->SetIntKey("excluded", info->excluded);
+  dict->SetStringKey("url", info->url);
+  dict->SetStringKey("provider", info->provider);
+  dict->SetStringKey("favicon_url", info->favicon_url);
+
+  Respond(TwoArguments(std::make_unique<base::Value>(result), std::move(dict)));
+}
+
+BraveRewardsGetPublisherPanelInfoFunction::
+    ~BraveRewardsGetPublisherPanelInfoFunction() {}
+
+ExtensionFunction::ResponseAction
+BraveRewardsGetPublisherPanelInfoFunction::Run() {
+  std::unique_ptr<brave_rewards::GetPublisherPanelInfo::Params> params(
+      brave_rewards::GetPublisherPanelInfo::Params::Create(*args_));
+  EXTENSION_FUNCTION_VALIDATE(params.get());
+
+  Profile* profile = Profile::FromBrowserContext(browser_context());
+  RewardsService* rewards_service =
+      RewardsServiceFactory::GetForProfile(profile);
+
+  if (!rewards_service) {
+    return RespondNow(NoArguments());
+  }
+
+  rewards_service->GetPublisherPanelInfo(
+      params->publisher_key,
+      base::Bind(
+          &BraveRewardsGetPublisherPanelInfoFunction::OnGetPublisherPanelInfo,
+          this));
+
+  return RespondLater();
+}
+
+void BraveRewardsGetPublisherPanelInfoFunction::OnGetPublisherPanelInfo(
+    const int32_t result,
+    std::unique_ptr<::brave_rewards::PublisherInfo> info) {
+  auto dict = std::make_unique<base::Value>(base::Value::Type::DICTIONARY);
+
+  if (!info) {
+    Respond(
+        TwoArguments(std::make_unique<base::Value>(result), std::move(dict)));
+    return;
+  }
+
+  dict->SetStringKey("publisher_key", info->id);
+  dict->SetStringKey("name", info->name);
+  dict->SetIntKey("percentage", info->percent);
+  dict->SetIntKey("status", info->status);
+  dict->SetIntKey("excluded", info->excluded);
+  dict->SetStringKey("url", info->url);
+  dict->SetStringKey("provider", info->provider);
+  dict->SetStringKey("favicon_url", info->favicon_url);
+
+  Respond(TwoArguments(std::make_unique<base::Value>(result), std::move(dict)));
+}
+
+BraveRewardsSavePublisherInfoFunction::
+    ~BraveRewardsSavePublisherInfoFunction() {}
+
+ExtensionFunction::ResponseAction
+BraveRewardsSavePublisherInfoFunction::Run() {
+  std::unique_ptr<brave_rewards::SavePublisherInfo::Params>
+      params(brave_rewards::SavePublisherInfo::Params::Create(
+          *args_));
+  EXTENSION_FUNCTION_VALIDATE(params.get());
+
+  Profile* profile = Profile::FromBrowserContext(browser_context());
+  RewardsService* rewards_service =
+      RewardsServiceFactory::GetForProfile(profile);
+
+  if (!rewards_service) {
+    return RespondNow(NoArguments());
+  }
+
+  rewards_service->SavePublisherInfo(
+      params->window_id,
+      params->media_type,
```

Comment:
should we maybe use `brave_rewards::ContentSite` and then in rewards service convert it to `ledger:PublisherInfo`?

Reply:
Makes sense, fixed.